multiqc/modules/bamutil/clipoverlap.py

```diff
@@ -0,0 +1,209 @@
+from multiqc.base_module import BaseMultiqcModule
+import re
+import logging
+
+log = logging.getLogger(__name__)
+
+
+class MultiqcModule(BaseMultiqcModule):
+    def __init__(self):
+        super().__init__(
+            name="BamUtil clipOverlap",
+            anchor="bamutil_clipoverlap",
+            href="http://genome.sph.umich.edu/wiki/BamUtil",
+            info="bamUtil is a repository that contains several programs that perform operations on SAM/BAM files. The clipOverlap submodule clips overlapping read pairs to prevent double-counting of coverage.",
+            doi="10.1186/s13059-014-0559-z",
+        )
+        self.bamutil_data = self.parse_logs()
+        if len(self.bamutil_data) == 0:
+            log.debug("Could not find any bamUtil clipOverlap reports")
+            raise UserWarning
+
+        # Write parsed data to a file
+        self.write_data_file(self.bamutil_data, "multiqc_bamutil_clipoverlap")
+
+        # Add software version placeholder
+        self.add_software_version(None)
+
+        self.bamutil_general_stats()
+
+    def parse_logs(self):
+        data = dict()
+        for f in self.find_log_files("bamutil_clipoverlap", filehandles=True):
+            s_name = self.clean_s_name(f["s_name"])
+
+            # Initialize all potential metrics to None
+            number_overlapping_pairs = None
+            avg_ref_bases_overlapped = None
+            variance_ref_bases = None
+            orientation_additional_clipping = None
+            forward_strand_clipped = None
+            reverse_strand_clipped = None
+            missing_overlapping_mates = None
```

Comment:
I think no need to set all these variables and then assign into the dict - can just assign directly and simplify the code.